auth/src/main/java/software/amazon/awssdk/auth/signer/Aws4Signer.java

```diff
@@ -537,11 +454,25 @@ private long generateExpirationDate(Date expirationDate) {
     private byte[] newSigningKey(AwsCredentials credentials,
                                  String dateStamp, String regionName, String serviceName) {
         byte[] kSecret = ("AWS4" + credentials.secretAccessKey())
-                .getBytes(Charset.forName("UTF-8"));
+            .getBytes(Charset.forName("UTF-8"));
         byte[] kDate = sign(dateStamp, kSecret, SigningAlgorithm.HmacSHA256);
         byte[] kRegion = sign(regionName, kDate, SigningAlgorithm.HmacSHA256);
         byte[] kService = sign(serviceName, kRegion,
                                SigningAlgorithm.HmacSHA256);
         return sign(SignerConstants.AWS4_TERMINATOR, kService, SigningAlgorithm.HmacSHA256);
     }
-}
+
+    public static class Builder  {
+        private static final Boolean DEFAULT_DOUBLE_URL_ENCODE = Boolean.TRUE;
+        private Boolean doubleUrlEncode = DEFAULT_DOUBLE_URL_ENCODE;
+
+        public Builder doubleUrlEncode(Boolean doubleUrlEncode) {
+            this.doubleUrlEncode = doubleUrlEncode;
+            return this;
+        }
```

Comment:
Any reason this is a signer property and not a params property? Would make it easier to use a signer across service clients, without the customer needing to know whether the specific service needs to double encode URLs.

Reply:
This value is constant for a signer irrespective of the request. So think its better to stay on the signer. As all signers expect S3 use doubleUrlEncode, there is no issue with using the Aws4Signer for all services. Ideally only advanced customers will be using the singer implementations directly and in that case they should be aware of the parameters in the signer.